ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -484,6 +484,43 @@ interface {{ ethernet_interface.name }}
 {%         if ethernet_interface.pim.ipv4.dr_priority is arista.avd.defined %}
    pim ipv4 dr-priority {{ ethernet_interface.pim.ipv4.dr_priority }}
 {%         endif %}
+{%         if ethernet_interface.poe.priority is arista.avd.defined %}
+   poe priority {{ ethernet_interface.poe.priority }}
+{%         endif %}
+{%         if ethernet_interface.poe.reboot.action is arista.avd.defined %}
+   poe reboot action {{ ethernet_interface.poe.reboot.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.link_down.action is arista.avd.defined %}
+{%             set poe_link_down_action_cli = 'poe link down action ' ~ ethernet_interface.poe.link_down.action %}
+{%             if ethernet_interface.poe.link_down.power_off_delay is arista.avd.defined %}
+{%                 set poe_link_down_action_cli = poe_link_down_action_cli ~ ' ' ~ ethernet_interface.poe.link_down.power_off_delay %}
+{%             endif %}
+   {{ poe_link_down_action_cli }}
+{%         endif %}
+{%         if ethernet_interface.poe.shutdown.action is arista.avd.defined %}
+   poe shutdown action {{ ethernet_interface.poe.shutdown.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.disabled is arista.avd.defined(true) %}
+   poe disabled
+{%         endif %}
+{%         if ethernet_interface.poe.limit.class is arista.avd.defined %}
+{%             set poe_class_map = dict({0: "15.40", 1: "4.00", 2: "7.00", 3: "15.40", 4: "30.00", 5: "45.00", 6: "60.00", 7: "75.00", 8: "90.00"}) %}
+{%             set poe_limit_cli = 'poe limit ' ~ poe_class_map[ethernet_interface.poe.limit.class] ~ ' watts' %}
+{%         elif ethernet_interface.poe.limit.watts is arista.avd.defined %}
+{%             set poe_limit_cli = 'poe limit ' ~ "%.2f" | format(ethernet_interface.poe.limit.watts | float) ~ ' watts' %}
```

Comment:
you are converting to float twice. Once with the filter and second with the format string. I think just doing `'poe limit ' ~ ethernet_interface.poe.limit.watts | float ~ ' watts'` should do it.

Reply:
Attempted your suggested solution and it looks like it fall short.  When converting `ethernet_interface.poe.limit.watts` from a string type to a float, if the the float value is specified to 1 decimal place or less, this value will print out to only 1 decimal place (i.e. When `45` is converted to a float, it displays `45.0` when printed).  This is an issue because EOS displays the watts value to 2 decimal places.  Therefore, we need to first convert our string value to a float value (`ethernet_interface.poe.limit.watts | float`) and then we can use the f-string expression `"%.2f"` (which only works on floats and will error out on strings) to format the float value to 2 decimal places.  So I believe we need the original expression `"%.2f" | format(ethernet_interface.poe.limit.watts | float)`